Extension/src/githubAPI.ts

```diff
@@ -181,8 +181,8 @@ export async function getTargetBuildInfo(updateChannel: string): Promise<BuildIn
  * @param updateChannel The user's updateChannel setting.
  * @return The Build if the user should update to it, otherwise undefined.
  */
-export function getTargetBuild(builds: Build[], userVersion: PackageVersion, updateChannel: string): Build | undefined {
-    if (!vscode.workspace.getConfiguration("extensions", null).get<boolean>("autoUpdate")) {
+export function getTargetBuild(builds: Build[], userVersion: PackageVersion, updateChannel: string, isFromSettingsChange: boolean): Build | undefined {
+    if (!isFromSettingsChange && !vscode.workspace.getConfiguration("extensions", null).get<boolean>("autoUpdate")) {
```

Comment:
If I understand correctly, this checks the `autoUpdate` setting only if not reacting to a setting change.  So, if the user changes from "Default" to "Insiders" we will now download and install an insiders, even if `autoUpdate` is disabled. (?)  Is that desirable?  Or, is there somewhere else this is short-circuited when `autoUpdate` is disabled?  (In 0.25.1, we had a bug in which autoUpdate was not honored: https://github.com/microsoft/vscode-cpptools/issues/3632 )

Reply:
I agree with Colen.  I don't think this is the desired behavior.  I think the only thing that should be allowed when autoUpdate is false would be downgrading when switching from "Insiders" to "Default".

Reply:
The "autoUpdate" is intended to be set to false for scenarios where the user doesn't want us automatically updating their versions without explicit consent/actions. When the user switches the channel to Insiders, they're essentially saying they want a one-time manual update to Insiders if one exists, but if autoUpdate is false, then they don't want future Insider updates unless they explicitly change the setting again. i.e. this allows users a mechanism to opt-in to updates on demand (pull updates) versus the current method of getting pushed updates whenever we happen to release. Is there some downside to this behavior I don't understand?

Reply:
In particular, autoUpdate is a VS Code setting that users may want to have set to false for other extensions to not auto-update, but would still like to have a mechanism to get the C/C++ Insiders.

Reply:
ok